### DISCUSSION TITLE
Items cannot be edited.

### DISCUSSION THREAD

**Discussion by NetCzech** [textbrowser/biblioteq#265] [Q&A] [answered]
Hello,
on a new computer, I ran the latest version of the program, and after connecting the existing database, I cannot edit the created book entries.

Error log:
2022/12/14 21:07:14,Database Error,Unable to retrieve the selected book's data.,Parameter count mismatch,Source\biblioteq_book.cc,981

How can I fix the problem? Thank you

**Comment by textbrowser** [comment]
I need more information. An image will help.

**Comment by NetCzech** [comment]
Double-clicking or pressing a button Modify Selected Item(s) will trigger the same action:

![BE](https://user-images.githubusercontent.com/76791894/207729063-60b9c6c2-2597-4550-8e0c-3acccaf2f1ec.jpg)

**Reply by textbrowser** [reply]
Thanks! :) Looking now.

**Comment by textbrowser** [comment]
Which version of BQ are you using? And was this SQLite database's schema upgraded?

**Comment by NetCzech** [comment]
After SQL Upgrade Schema I get this message:

![SQL](https://user-images.githubusercontent.com/76791894/207815157-40738c31-9590-476b-aaba-422e2a004421.png)

BiblioteQ Version: 2022.11.25

**Comment by textbrowser** [comment]
Can you access items now?

Newer versions of BQ remind SQLite people to upgrade. I archived much of the upgrade statements many months ago (March, 2022) so your database schema may be flawed.

Anyway, if it still doesn't work, upload it here and I'll review it.

**Reply by NetCzech** [reply]
[My book database](https://webshare.cz/#/file/9u75KsAwfU), thank you.

**Reply by textbrowser** [reply]
Cannot download it without paying and I'm not creating an account.

**Reply by textbrowser** [reply]
You can create a repository on GH and upload the file there.

**Reply by NetCzech** [reply]
> You can create a repository on GH and upload the file there.

It´s free download server. Choose to download without VIP.

![00](https://user-images.githubusercontent.com/76791894/208192167-a32a664c-0dc2-44cf-ba7d-6ef7b9dada95.png)

**Reply by textbrowser** [reply]
OK! The site declined this option. Now it's working.

**Reply by textbrowser** [reply]
It's important that you update BQ and your database(s). I may not be able to correct it. I also don't maintain all previous update statements.

**Reply by textbrowser** [reply]
Yeah, your database is busted.

**Reply by textbrowser** [reply]
When did you last update BQ?

**Reply by NetCzech** [reply]
> When did you last update BQ?

Maybe a year ago :)

**Reply by textbrowser** [reply]
Well, BQ doesn't update itself. All database changes are published in release notes with many, many reminders. I corrected it but I should not assist you. :P

**Reply by textbrowser** [reply]
https://textbrowser.github.io/biblioteq/

March 30, 2022. Archived SQLite upgrade statements.

**Reply by textbrowser** [reply]
https://github.com/textbrowser/biblioteq/releases/tag/2022.12.25

Let me know that you've downloaded the database so I can remove the file.

**Reply by NetCzech** [reply]
> https://github.com/textbrowser/biblioteq/releases/tag/2022.12.25
> 
> Let me know that you've downloaded the database so I can remove the file.

It works perfectly. Thank you very much.

**Comment by meteos77** [comment]
if only one base : make a blank base and do the merge :
demo provided by textbrowser : https://github.com/textbrowser/biblioteq/discussions/191

for several bases : not sure

**Comment by textbrowser** [comment, marked as answer]
Corrected by rebuilding the software with archived SQLite statements and upgrading the database.
